lib/rules/no-nested-tests.js

```diff
@@ -0,0 +1,38 @@
+'use strict';
+
+var astUtils = require('../util/ast');
+
+module.exports = function noNestedTests(context) {
+    var testNestingLevel = 0;
+
+    function report(callExpression) {
+        var message = astUtils.isTestCase(callExpression) ? 'Unexpected test nested within another test.' :
```

Comment:
*slight* boost, you call have an argument `isTestCase` for this function, as you have already computed in the calling function.